sources/Algo/include/Algo.h

```diff
@@ -259,5 +260,101 @@ class LoadDefinitionAlgorithm : public NodeAlgorithm {
+   * 
+   * @return status of the comparison
+   */
+  bool operator==(const HvdcLineDefinition& other) const;
+
+  /**
+   * @brief Constructor
+   *
+   * @param id the HvdcLine id
+   * @param converterType type of converter of the hvdc line
+   * @param converter1_id first converter id 
+   * @param converter1_busId first converter bus id
+   * @param converter1_voltageRegulationOn firt converter voltage regulation parameter, for VSC converters only
+   * @param converter2_id second converter id 
+   * @param converter2_busId second converter bus id
+   * @param converter2_voltageRegulationOn second converter voltage regulation parameter, for VSC converters only
+   * @param position position of the converters of this hvdc line compared to the main connex component
+   */
+  explicit HvdcLineDefinition(const HvdcLineId& id, const inputs::HvdcLine::ConverterType converterType, const ConverterId& converter1_id,
+                              const BusId& converter1_busId, const boost::optional<bool>& converter1_voltageRegulationOn, const ConverterId& converter2_id,
+                              const BusId& converter2_busId, const boost::optional<bool>& converter2_voltageRegulationOn, const Position position) :
+      id{id},
+      converterType{converterType},
+      converter1_id{converter1_id},
+      converter1_busId{converter1_busId},
+      converter1_voltageRegulationOn{converter1_voltageRegulationOn},
+      converter2_id{converter2_id},
+      converter2_busId{converter2_busId},
+      converter2_voltageRegulationOn{converter2_voltageRegulationOn},
+      position{position} {}
+
+  const HvdcLineId id;                                         ///< HvdcLine id
+  const inputs::HvdcLine::ConverterType converterType;         ///< type of converter of the hvdc line
+  const ConverterId converter1_id;                             ///< first converter id
+  const BusId converter1_busId;                                ///< first converter bus id
+  const boost::optional<bool> converter1_voltageRegulationOn;  ///< firt converter voltage regulation parameter, for VSC converters only
+  const ConverterId converter2_id;                             ///< second converter id
+  const BusId converter2_busId;                                ///< second converter bus id
+  const boost::optional<bool> converter2_voltageRegulationOn;  ///< second converter voltage regulation parameter, for VSC converters only
+  Position position;                                           ///< position of the converters of this hvdc line compared to the main connex component
+};
+
+/**
+ * @brief the controller interface definition algorithm
+ */
+class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+ public:
+  using HvdcLineId = std::string;  ///< HvdcLine id definition
+  using HvdcLineMap =
+      std::map<HvdcLineId, HvdcLineDefinition>;  ///< HvdcLineMap definition, useful in algo.cpp for checking if a hvdcLineDefinition is already inserted
```

Comment:
innacurate comment: the map is the output of the algorithm and is provided by the context

Reply:
So what should I write ?

Reply:
generic comment, like 'Alias for map of hvdc line definition"